control/control_loop.go

```diff
@@ -195,7 +197,11 @@ func (l *Loop) SetConfigAt(ctx context.Context, name string, config BlockConfig)
 	if !ok {
 		return errors.Errorf("cannot return Config for nonexistent %s", name)
 	}
-	return blk.blk.UpdateConfig(ctx, config)
+	if err := blk.blk.UpdateConfig(ctx, config); err != nil {
+		return err
+	}
+	l.cfg.Blocks[blk.blockIndex] = blk.blk.Config(ctx)
```

Comment:
block index used here because `UpdateConfig` was not actually updating the config it was just updating the blocks , and the blocks aren't a map so I can't just use the name

Reply:
wait no other way. it updates the config but not the actual blocks